src/main/java/com/casper/sdk/helper/TransactionHelper.java

```diff
@@ -18,38 +30,78 @@
  */
 public class TransactionHelper {
 
+
+    @SuppressWarnings({"unchecked", "rawtypes", "OptionalOfNullableMisuse"})
+    public static <TransferTarget> TransactionV1 newTransfer(final BigInteger amount,
+                                                             final URef maybeSource,
+                                                             final TransferTarget target,
+                                                             final Long maybeId) throws ValueSerializationException {
+        final List<NamedArg<?>> namedArgs = new ArrayList<>();
+        namedArgs.add(NamedArg.builder().type("amount").clValue((AbstractCLValue) new CLValueU512(amount)).build());
+        namedArgs.add(NamedArg.builder()
+                .type("source")
+                .clValue((AbstractCLValue) new CLValueOption(Optional.ofNullable(new CLValueURef(maybeSource))))
+                .build()
+        );
+
+        final AbstractCLValue clTarget;
+
+        if (target instanceof URef) {
+            clTarget = new CLValueURef((URef) target);
+        } else if (target instanceof PublicKey) {
+            clTarget = new CLValuePublicKey((PublicKey) target);
+        } else if (target instanceof Digest) {
+            clTarget = new CLValueByteArray(((Digest) target).getDigest());
+        } else {
+            throw new IllegalArgumentException("Invalid target type " + target);
+        }
+
+        namedArgs.add(NamedArg.builder().type("target").clValue(clTarget).build());
+
+        if (maybeId != null) {
+
+            namedArgs.add(new NamedArg<>("id", new CLValueOption(
+                    Optional.ofNullable(maybeId != null ? new CLValueU64(BigInteger.valueOf(System.currentTimeMillis())) : null)
+            )));
+        }
+
+        return TransactionV1.builder()
+                //          .payload(buildTransactionPayload(initiatorAddr, new Date(), ttl, chainName, pricingMode))
```

Comment:
remove the commented out code

Reply:
It is still WIP will change to use new code but test will fail

Reply:
Resolved